src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIRequestSmelter.java

```diff
@@ -478,11 +492,18 @@ private IAIState fillUpFurnace()
                                 return getState();
                             }
                             worker.getCitizenItemHandler().hitBlockWithToolInHand(walkTo);
-                            job.setProgress(job.getProgress() + toTransfer);
-                            InventoryUtils.transferXOfFirstSlotInItemHandlerWithIntoInItemHandler(
-                            worker.getInventoryCitizen(), smeltable, toTransfer,
-                            new InvWrapper(furnace), SMELTABLE_SLOT);
-                        }
+                            int transferred = InventoryUtils.transferXOfFirstSlotInItemHandlerWithIntoInItemHandler(
+                                worker.getInventoryCitizen(), smeltable, toTransfer,
+                                new InvWrapper(furnace), SMELTABLE_SLOT);
+                            // It's possible we have less in the stack than we want to transfer, so transfer what we can, and try one more time. 
+                            if (transferred < toTransfer)
+                            {
+                                transferred += InventoryUtils.transferXOfFirstSlotInItemHandlerWithIntoInItemHandler(
+                                    worker.getInventoryCitizen(), smeltable, toTransfer - transferred,
+                                    new InvWrapper(furnace), SMELTABLE_SLOT);
+                            }
+                            job.setProgress(job.getProgress() + transferred);
```

Comment:
formatting seems off here.

Reply:
How can this fail and need to be executed twice?

Reply:
You want to transfer 20, the first slot it finds has 8, the rest are in the second slot. It only works with one slot, so the first request is for 20 on the slot with 8. You need to a follow request on the slot with 12 to complete it.

Reply:
Then we should probably use a better one.

Reply:
Ok, that's more consistent formatting now.